### PR TITLE
Add Dividers into Navigation Sidebar

### PR DESCRIPTION
**Pull request by laurapacilio**
### What
We need to add dividers into the highest level docs sidebar for the devdot launch. This PR adds sidebar dividers to:
- Group elements into logical categories
- Differentiate product names (Terraform Cloud) from parts of the documentation that let you accomplish a specific task (Develop Plugins)

### Why
Required for devdot launch.

#### Before Edits
Screen Shot 2022-08-31 at 5 34 07 PM

#### After Edits
Screen Shot 2022-08-31 at 6 48 01 PM

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [ ] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [ ] Description links to related pull requests or issues, if any.

#### Content
- [ ] Redirects have been added for moved, renamed, or deleted pages. This requires a separate PR in the [`terraform-website` repository](https://github.com/hashicorp/terraform-website) `redirects.next.js` file.
- [ ] API documentation and the API Changelog have been updated. 
- [ ] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [ ] Pages with related content are updated and link to this content when appropriate.
- [ ] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [ ] New pages have metadata (page name and description) at the top.
- [ ] New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [ ] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [ ] UI elements (button names, page names, etc.) are bolded.
- [ ] The Vercel website preview successfully deployed.

#### Reviews
- [ ] I or someone else reviewed the content for technical accuracy.
- [ ] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

File: website/data/docs-nav-data.json [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
-    "title": "Integration Program",
-    "path": "partnerships"
-  },
-  { "title": "Terraform Tools", "path": "terraform-tools" },
   { "title": "CDK for Terraform", "href": "/cdktf" },
+  { "divider": true },
+  { "title": "Use Providers", "href": "/language/providers" },
+  { "title": "Develop Plugins", "href": "/plugin" },
+  { "title": "Publish to the Registry", "href": "/registry" },
+  { "divider": true },
```
Comment (laurapacilio) [inline website/data/docs-nav-data.json:13]:
```suggestion
  { "divider": true },
    {
    "title": "Integration Program",
    "path": "partnerships"
  },
```

File: website/data/docs-nav-data.json [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
   { "title": "Terraform Enterprise", "href": "/enterprise" },
-  { "title": "Provider Use", "href": "/language/providers" },
-  { "title": "Plugin Development", "href": "/plugin" },
-  { "title": "Registry Publishing", "href": "/registry" },
-  {
+  { "title": "CDK for Terraform", "href": "/cdktf" },
+  { "divider": true },
+  { "title": "Use Providers", "href": "/language/providers" },
+  { "title": "Develop Plugins", "href": "/plugin" },
+  { "title": "Publish to the Registry", "href": "/registry" },
```
Comment (laurapacilio) [inline website/data/docs-nav-data.json:12]:
```suggestion
  { "title": "Provider Use", "href": "/language/providers" },
  { "title": "Plugin Development", "href": "/plugin" },
  { "title": "Registry Publishing", "href": "/registry" },
```

File: website/data/docs-nav-data.json [hunk 3]
```diff
@@ -1,17 +1,20 @@
 [
   { "title": "Intro to Terraform", "href": "/intro" },
+  { "divider": true },
   { "title": "Configuration Language", "href": "/language" },
-  { "title": "Terraform CLI", "href": "/cli" },
+  { "title": "CLI", "href": "/cli" },
```
Comment (judithpatudith) [inline website/data/docs-nav-data.json:5]:
```suggestion
  { "title": "Terraform CLI", "href": "/cli" },
```

**Review by judithpatudith — APPROVED** [review]
I'd usually object to putting Terraform in front of everything but in this one case it CLI really looks weird with the other editions. Otherwise LGTM

**Pull request merged**